src/app/data-model-provider/Provider.h

```diff
@@ -98,8 +100,9 @@ class Provider : public ProviderMetadataTree
     ///    - data (as CommandDataIB) which is assumed a "response as a success"
     ///    - status (as a CommandStatusIB) which is considered a final status, usually an error however
     ///      cluster-specific success statuses also exist.
-    virtual ActionReturnStatus Invoke(const InvokeRequest & request, chip::TLV::TLVReader & input_arguments,
-                                      CommandHandler * handler) = 0;
+    ///
+    virtual std::optional<ActionReturnStatus> Invoke(const InvokeRequest & request, chip::TLV::TLVReader & input_arguments,
```

Comment:
nit, can you update documentation? Please mention that if there is an error return that the handler has not sent anything in response to the command